controllers/job_mgr/job_mgr_controller.go

```diff
@@ -0,0 +1,173 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package job_mgr
+
+import (
+	"context"
+	"fmt"
+
+	workloadv1alpha1 "github.com/trustyai-explainability/trustyai-service-operator/api/lmes/v1alpha1"
+	"github.com/trustyai-explainability/trustyai-service-operator/controllers/lmes"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/podset"
+)
+
+// +kubebuilder:rbac:groups=scheduling.k8s.io,resources=priorityclasses,verbs=list;get;watch
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;watch;update;patch
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloads/finalizers,verbs=update
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=resourceflavors,verbs=get;list;watch
+// +kubebuilder:rbac:groups=kueue.x-k8s.io,resources=workloadpriorityclasses,verbs=get;list;watch
+
+type LMEvalJob struct {
+	workloadv1alpha1.LMEvalJob
+}
+
+func ControllerSetUp(mgr manager.Manager, ns, configmap string, recorder record.EventRecorder) error {
+	ctx := context.TODO()
+	if err := jobframework.SetupWorkloadOwnerIndex(ctx, mgr.GetFieldIndexer(), workloadv1alpha1.GroupVersion.WithKind("LMEvalJob")); err != nil {
+		return fmt.Errorf("workload indexer: %w", err)
+	}
+	lmes.Job_mgr_enabled = true
+	return jobframework.NewGenericReconcilerFactory(
+		func() jobframework.GenericJob { return &LMEvalJob{} },
+		func(b *builder.Builder, c client.Client) *builder.Builder {
+			return b.Named("LMEvalJobWorkload")
+		},
+	)(mgr.GetClient(), mgr.GetEventRecorderFor("kueue")).SetupWithManager(mgr)
+
+}
+
+// Object returns the job instance.
+func (job *LMEvalJob) Object() client.Object {
+	return &job.LMEvalJob
+}
+
+// IsSuspended returns whether the job is suspended or not.
+func (job *LMEvalJob) IsSuspended() bool {
+	return job.Spec.Suspend
+}
+
+func (job *LMEvalJob) Suspend() {
+	job.Spec.Suspend = true
+}
+
+// RunWithPodSetsInfo will inject the node affinity and podSet counts extracting from workload to job and unsuspend it.
+func (job *LMEvalJob) RunWithPodSetsInfo(podSetsInfo []podset.PodSetInfo) error {
+	job.Spec.Pod.Affinity = convertToAffinity(podSetsInfo)
+	job.Spec.Suspend = false
```

Comment:
one question: so this function needs you to update the `job.Spec.Suspend` to `false` but no need to set the `job.Spec.Suspend` to `true` in the `RestorePodSetsInfo` function?

Reply:
I wondered that too. I am not sure what triggers this RestorePodSetsInfo function. I will add it for now.

Reply:
I observed RestorePodSetsInfo is called right before a job preempted But if I had job.Spec.Suspend = false the job would not be evicted. So I removed it again.